src/index.js

```diff
@@ -95,27 +95,24 @@ class ESLintWebpackPlugin {
         return;
       }
 
-      const seen = new Set();
-
       // Gather Files to lint
       compilation.hooks.finishModules.tap(ESLINT_PLUGIN, (modules) => {
-        const resources = Array.from(modules)
-          // @ts-ignore
-          .filter((module) => module.resource)
-          // @ts-ignore
-          .map(({ resource }) => resource.split('?')[0])
-          .filter(
-            (resource) =>
-              resource &&
-              !seen.has(resource) &&
-              isMatch(resource, wanted) &&
-              !isMatch(resource, exclude)
-          );
-
-        resources.forEach((resource) => seen.add(resource));
-
-        if (resources.length > 0) {
-          lint(resources);
+        const resources = new Set(
+          Array.from(modules)
+            // @ts-ignore
+            .filter((module) => module.resource)
+            // @ts-ignore
+            .map(({ resource }) => resource.split('?')[0])
+            .filter(
+              (resource) =>
+                resource &&
+                isMatch(resource, wanted) &&
+                !isMatch(resource, exclude)
+            )
+        );
```

Comment:
Recreating `modules` from `Array` and when convert it to `Set` is big perf problem, just is `for (const module of modules) { }` fir the best perf

Reply:
Updated using spread operator

Reply:
WDTY? https://github.com/webpack-contrib/eslint-webpack-plugin/pull/73/commits/79d6aba24c0332f98904c28ea03da77f92ac6db0#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556